### ISSUE TITLE
Migrate libblockdev from libparted to libfdisk

### ISSUE THREAD

**Issue by squimrel** [storaged-project/libblockdev#245]
Basically libparted is way to smart for libblockdev.

For example libblockdev [only knows two table types](https://github.com/storaged-project/libblockdev/blob/dbe13d5ca58d698b340a268f5ce3556e56e07616/src/plugins/part.c#L176).

Where as [libparted knows a lot more](https://git.savannah.gnu.org/cgit/parted.git/tree/libparted/labels) even things like the mac label.

Which is not necessarily a good thing. For example I ran into a specific issue where the [isohybrid](http://www.syslinux.org/wiki/index.php?title=Isohybrid) layout is treated as a mac label and therefore I couldn't add a fourth partition to it because a mac label may only have a maximum of three partitions.
Basically in my use-case I'd only like to modify the MBR partition table (The isohybrid layout consists of multiple partition tables)

[libfdisk on the other hand is much simpler](https://github.com/karelzak/util-linux/tree/master/libfdisk/src). It only knows the bsd and sun label apart from the dos and mac label.

It also seems like this project could benefit from libfdisk since in libfdisk you can specify the partition number and use default values determined by libfdisk for start and end if you want to.

Apparently I poorly don't seem to have time to work on this myself at the moment but I would really appreciate it if you would work on this.

**Comment by MartinBriza** [comment]
Please also include a concrete example of a use case when this is a problem. Like what kind of media you use and what are you trying to achieve with it using UDisks

**Comment by squimrel** [comment]
We're trying to add a fourth partition on a flash drive that already has a Fedora image on it. We'd like to use this fourth partition to store an overlay file that's used for persistent storage when booted into the live system.

**Comment by squimrel** [comment]
By the way it would be reasonable to argue that trying manipulate partitions on an isohybrid layout is nonsense since the tools I know of only modify one partition table at the time and isohybrid has three partition tables on it.

**Comment by vpodzime** [comment]
> ... isohybrid has three partition tables on it.

Yes, but aren't two of them basically just "masks" ensuring that the device is bootable on various platforms? Is there any specification of which of the partition tables to actually respect as a partitioning schema of the device?

**Comment by squimrel** [comment]
> aren't two of them basically just "masks"

I'm not sure if all the partition tables specify the same partitions.

> Is there any specification of which of the partition tables to actually respect as a partitioning schema of the device?

In case you're asking whether there's a specification that specifies which partition table should be used if there're multiple present then I don't know if there's one. I'd guess that there isn't. Linux seems to prefer the MBR partition table.
In addition to that. All of the partition tables are valid in the sense that libparted would understand all of them separately for example. If I didn't answer your question then I didn't get your question.

**Comment by andhe** [comment]
On a semi-related note (maybe this should be a separate issue?), it would be nice if the 'part' plugin where ported away from spawning gdisk. AFAIK gdisk is pretty unmaintained and these days (util-linux) fdisk supports GPT (since a long time now). Simply switching from (s)gdisk to (s)fdisk would not be as good as not spawning cli commands at all, by moving to a proper library like libfdisk (or libparted, or whatever).

**Comment by vpodzime** [comment]
> On a semi-related note (maybe this should be a separate issue?)

Yes, please create a separate issue for this.

**Comment by vojtechtrefny** [comment]
We are currently (slowly) working on rewriting the part plugin to libfdisk on a separate branch, see for example #407

**Comment by vpodzime** [comment]
Thanks @vojtechtrefny for finishing this up!!! :tada: :1st_place_medal:
